provide/Services/NatsClient.cs

```diff
@@ -0,0 +1,120 @@
+
+        public NatsClient(string token)
+        {
+            this.token = token;
+        }
+
+        public NatsClient(string host, string path, string scheme, string token)
+        {
+            this.host = host;
+            this.path = path;
+            this.scheme = scheme;
+            this.token = token;
+            this.natsUrl = $"{scheme}://${host}/${path}/";
+        }
+
+        public void Connect()
+        {
+            var opts = ConnectionFactory.GetDefaultOptions();
+            SetJWTAndNKeyHandlers(opts);
+            // if not provided it will use default local server url
+            opts.Url = this.natsUrl;
+            // add try catch?
+            this.connection = new ConnectionFactory().CreateConnection(opts);
+        }
+
+        public ConnState GetConnectionState()
+        {
+            return this.connection.State;
+        }
+
+        public void Close()
+        {
+            if (this.connection.IsClosed()) 
+            {
+                return;
+            }
+            // should we do drain instead?
+            this.connection.Close();
+        }
+
+        public int GetSubscriptionCount() 
+        {
+            return this.connection.SubscriptionCount;
+        }
+
+        public void Publish(string subject, string reply, byte[] payload)
+        {
+            this.connection.Publish(subject, reply, payload);
+        }
+
+        public void Publish(string subject, byte[] payload)
+        {
+            this.connection.Publish(subject, payload);
+        }
+
+        public IAsyncSubscription Subscribe(string subject, EventHandler<MsgHandlerEventArgs> msgHandler)
+        {
+            var subscription = this.connection.SubscribeAsync(subject, msgHandler);
+            this.subscriptions.Add(subject, subscription);
+            return subscription;
+        }
+
+        // it looks like this is the only way to unsubcribe from specific subscription
+        // this method might be redundant but keeping it just to have all functionalities
+        public void Unsubscribe(IAsyncSubscription subscription)
+        {
+            subscription.Unsubscribe();
+        }
+
+        // might not be needed
+        public void Unsubscribe(string subject)
+        {
+            IAsyncSubscription subscription;
+            if (this.subscriptions.TryGetValue(subject, out subscription) && subscription != null)
+            {
+                subscription.Unsubscribe();
+            }
+            else 
+            {
+                throw new NATSBadSubscriptionException("Can not find subscription to unsubscribe");
+            }
+        }
+
+        public async Task<Msg> RequestAsync(string subject, int timeout, byte[] payload)
+        {
+            return await this.connection.RequestAsync(subject, payload, timeout);
+        }
+        
+        private void SetJWTAndNKeyHandlers(Options opts)
```

Comment:
We aren't leveraging the `NKey` pattern at this time so this could probably stand to be renamed slightly.